app/models/working_hour.rb

```diff
@@ -69,4 +71,11 @@ def close_after_open
 
     errors.add(:close_hour, 'Closing time cannot be before opening time')
   end
+
+  def update_open_all_day_hours
+    return unless open_all_day?
+
+    WorkingHour.find_by(day_of_week: day_of_week, open_all_day: false)&.update(closed_all_day: false, open_hour: 0, open_minutes: 0, close_hour: 23,
```

Comment:
This will iterate over all the WorkingHour records in the DB. 

better to do 

```
before_validation :ensure_open_all_day_hours

def ensure_open_all_day_hours
   return unless open_all_day? 
  
   self.closed_all_day = false
   self.open_hour = 0
   self.open_minutes = 0
   self.close_hour = 23
   sefl.close_minutes = 59
end


```

We could also add a xor validation for closed_all_day & open_all_day to ensure both doesn't turn to be true.